SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -4265,74 +4267,175 @@ void CSolver::LoadInletProfile(CGeometry **geometry,
           /*--- Get data for this profile. ---*/
 
           vector<passivedouble> Inlet_Data = profileReader.GetDataForProfile(jMarker);
-
           unsigned short nColumns = profileReader.GetNumberOfColumnsInProfile(jMarker);
 
-          vector<su2double> Inlet_Values(nColumns);
+          /*--- Define Inlet Values vectors before and after interpolation (if needed) ---*/
+          vector<su2double> Inlet_Values(nCol_InletFile);
+          vector<su2double> Inlet_Interpolated(nColumns);
 
-          /*--- Loop through the nodes on this marker. ---*/
+          unsigned long nRows = profileReader.GetNumberOfRowsInProfile(jMarker);
+          /*--- Sort Data Column wise for interpolation ---*/
+          vector<su2double> Inlet_Columns (nRows);
+          vector<su2double> Inlet_Radii (nRows);
 
-          for (iVertex = 0; iVertex < geometry[MESH_0]->nVertex[iMarker]; iVertex++) {
+          switch(config->GetKindInletInterpolationFunction()){
 
-            iPoint   = geometry[MESH_0]->vertex[iMarker][iVertex]->GetNode();
-            Coord    = geometry[MESH_0]->node[iPoint]->GetCoord();
-            min_dist = 1e16;
+            case (NONE):
+            Interpolate = false;
+            break;
 
-            /*--- Find the distance to the closest point in our inlet profile data. ---*/
+            case (ONED_AKIMASPLINE_SPANWISE):
+            case (ONED_LINEAR_SPANWISE):
+            Interpolate = true;
+
+            cout<<"Switch statement confirmed, printing Interpolate bool: "<<Interpolate<<endl;
+
+            if(config->GetKindInletInterpolationFunction() == ONED_LINEAR_SPANWISE)
+              Interpolation_Function = "Linear";
+            else if (config->GetKindInletInterpolationFunction() == ONED_AKIMASPLINE_SPANWISE)
+              Interpolation_Function = "Akima";
+              
+              /* --- Creating and initializing a new pointer to object for class Interpolation ---*/
+              CInletInterpolation **s = new CInletInterpolation*[nRows];
+              for (int i=0;i<=nColumns;i++)
+                s[i] = new CInletInterpolation[nRows];
```

Comment:
I need to make the interpolation function for every column of the inlet file. So if there are 6 columns in the file I need 6 separate, say, Akima Interpolation functions. Every "s [ i ]" points to the respective column. That's why I need a 2D matrix.

Otherwise, I would need to create separate objects for every column which I would not be sure of as the number of columns may vary between files.

Reply:
I just had a look at the profile reader, for each marker it returns a vector of size nRows * nColums right? which means you do not need to interpolate each column independently, before you changed the code you were operating on the entire profile, keep that feature as it is the right way to do it IMO, there is not need to create extra copies of the data to separate things.

Reply:
Previously I was calculating the interpolated value for every vertex at that instant the vertex is called. So I took that point and some points before and after it in the same column to calculate the interpolated value. Now I am using "Set" and "Eval" functions which need to have the different splines "Set" before it starts to loop over the vertexes when it uses "Eval".

An alternative is to pass the Inlet_Data (from profileReader) to C1DInterpolation class and then do the separation there. But I would need interpolation functions equal to the number of columns that I have.